src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.dao.ShoppingCartDao;
+import com.dev.cinema.dao.TicketDao;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.MovieSession;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        ShoppingCart shoppingCart = getByUser(user);
+        Ticket ticket = getTicket(movieSession, user);
+        shoppingCart.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user);
+    }
+
+    @Override
+    public void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartDao.add(shoppingCart);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.getTickets().clear();
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    private Ticket getTicket(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        return ticketDao.add(ticket);
```

Comment:
Don't see the point of making this private method since you are using this only once

Reply:
but okey

Reply:
the same, but checkstyle didn't think the same